### PR TITLE
Combine unread notifications into a single one

### PR DESCRIPTION
**Pull request by xPaw**
Turns this:
photo_2017-08-28_15-35-55

Into this:

### CODE REVIEW

**Comment by astorije** [comment]
Will there be a way to apply this same code for regular desktop notifications as well?

**Comment by xPaw** [comment]
@astorije Perhaps in the future, but desktop notifications expire automatically so it's less of a problem.

**Comment by astorije** [comment]
Fair. Don't push notifications also expire on desktop though? I need to play with this to get a better sense of what happens then.

File: src/plugins/irc-events/message.js [hunk 1]
```diff
@@ -109,16 +109,14 @@ module.exports = function(irc, network) {
 			let title = data.nick;
 
 			if (chan.type !== Chan.Type.QUERY) {
-				title += ` (${chan.name}) mentioned you`;
-			} else {
-				title += " sent you a message";
+				title += ` (${chan.name})`;
```
Comment (astorije) [inline src/plugins/irc-events/message.js]: Any chance you can/want to use the same wording than for desktop notifications? "says:", etc.

**Comment by astorije** [comment]
So desktop notifications do expire. It's a bit weird on laptop, but oh well.
Left a comment about consistent wording with desktop notifications though, what do you think @xPaw?

File: src/plugins/irc-events/message.js (hunk 1 above)
Comment (xPaw) [reply src/plugins/irc-events/message.js]: The shorter it is, the better, imo. The way it groups messages looks a little messy, so we might something like bullet points.

Comment (astorije) [reply src/plugins/irc-events/message.js]: Alright then.

**Comment by xPaw** [comment]
I ran this PR for a couple of days myself now, and found it was a subpar experience. So now I group notifications per channel/target and count mentions correctly on the server (it will say `1 new message` for queries and `1 mention` for channels).

It will also only display latest message per notification instead of combining them on new lines (it was really busy and hard to read).

**Comment by astorije** [comment]
When receiving multiple messages from a same person, if you only display the latest one, it's lacking context and is therefore not really helpful. What about showing the message if only one or "X new messages" if more than one (what telegram does), or another could be "<first message> and X other messages" type of thing.

**Comment by xPaw** [comment]
@astorije I have a hard time understanding your last sentence, but the implementation does mirror what Telegram does as far as I know.

**Comment by astorije** [comment]
> or another could be " and X other messages" type of thing.

Alright so what I meant is that the combined notification popup could be:

> [First of the N messages received]
> *... and [N-1] other messages*

Better now?

**Comment by xPaw** [comment]
@astorije Would N-1 be just highlights/mentions or all unread messages for that channel?

**Comment by astorije** [comment]
Just what would usually trigger notification popups, so I guess only highlights/mentions.

**Comment by xPaw** [comment]
Added that.

File: src/models/chan.js [hunk 2]
```diff
@@ -25,7 +25,7 @@ function Chan(attr) {
 		type: Chan.Type.CHANNEL,
 		firstUnread: 0,
 		unread: 0,
-		highlight: false,
+		highlight: 0,
```
Comment (astorije) [inline src/models/chan.js]: Kind of a nitpick, but could you rename this into `highlights` or `highlightCount` then?

Comment (xPaw) [reply src/models/chan.js]: This requires finding all the instances (client included), I'd rather not.

Comment (astorije) [reply src/models/chan.js]: Apart from those you have modified in this PR, the only other occurrence of this that I found is https://github.com/thelounge/lounge/blob/master/client/views/chan.tpl#L3

Comment (xPaw) [reply src/models/chan.js]: There is definitely more.

Comment (astorije) [reply src/models/chan.js]: All others that I found are for `Msg.highlight`.

**Pull request merged**